I.MX6ULL secondary boot (part 1)

igoropaniuk opened the pull request:
Doesn't support switching to secondary boot path yet.

igoropaniuk committed:
[FIO internal] imx: secondary_boot: replace call_imx_sip with arm_smccc_smc

Replace legacy SMC call_imx_sip wrapper with the generic one arm_smccc_smc.

Signed-off-by: Igor Opaniuk <igor.opaniuk@foundries.io>

igoropaniuk committed:
[FIO internal] mx6: PERSIST_SECONDARY_BOOT handling

Add functions for retrieving/setting PERSIST_SECONDARY_BOOT bit
in SRC GPR10.

Signed-off-by: Igor Opaniuk <igor.opaniuk@foundries.io>

igoropaniuk committed:
[FIO internal] imx: move secondary boot related symbols

Move SECONDARY_BOOT_RUNTIME_DETECTION and SECONDARY_BOOT_SECTOR_OFFSET
symbols to the common iMX Kconfig. Extend support for iMX6UL/ULL.

Signed-off-by: Igor Opaniuk <igor.opaniuk@foundries.io>

igoropaniuk committed:
[FIO internal] imx: mmc: detect offset of next image in runtime

Detect offset of the next boot image in runtime for all iMX-based SoCs
if SECONDARY_BOOT_RUNTIME_DETECTION and SECONDARY_BOOT_SECTOR_OFFSET
are set.

Signed-off-by: Igor Opaniuk <igor.opaniuk@foundries.io>

Tim-Anderson reviewed (approved):
LGTM

ricardosalveti commented on arch/arm/mach-imx/imx_secondary_boot.c:
```diff
@@ -100,3 +102,4 @@ U_BOOT_CMD(
 	"Assert WDOG Software Reset Signal (internal reset) via TF-A",
 	"\n"
 );
+
```
Cosmetic, no need for the empty line here.

igoropaniuk replied:
sure, will fix

ricardosalveti commented:
Looks good, just one minor comment (cosmetics).

igoropaniuk committed:
[FIO internal] fastboot: fsl: add sit offset for imx6ul/imx6ull

Add Secondary Image Table offset for i.MX6UL/i.MX6ULL, which is used
in redundant boot setups.

Signed-off-by: Igor Opaniuk <igor.opaniuk@foundries.io>

igoropaniuk commented:
added one additional commit

ricardosalveti commented:
> added one additional commit

Missing [fio] tag on this one.

igoropaniuk commented:
@ricardosalveti fixed

ricardosalveti reviewed (approved):
LGTM
